add minheap.c

leviongit opened the pull request.

leviongit committed:
add minheap.c

kfischer-okarin commented on minheap.c:
```diff
@@ -0,0 +1,242 @@
+static mrb_sym lt_sym;
+
+minheap_t *minheap_new(mrb_state *mrb, uint8_t layers) {
+  assert(layers <= 63);
```
Why? 

It would be nice to either

- a) have a comment that explains why this precondition is important

or even better

- b) extract the 63 into a constant that is named so clearly that you need no more comments

leviongit committed:
style fixes and nitpicks

style: extract magic constant into magic macro
nitpick: cast `1` to `size_t` (for good measure)
